packages/block-library/src/missing/edit.native.js

```diff
@@ -35,6 +35,13 @@ import styles from './style.scss';
 // Blocks that can't be edited through the Unsupported block editor identified by their name.
 const UBE_INCOMPATIBLE_BLOCKS = [ 'core/block' ];
 const I18N_BLOCK_SCHEMA_TITLE = 'block title';
+const CUSTOM_UNSUPPORTED_BLOCK_MESSAGE = {
+	'core/gallery': {
+		/* translators: Unsupported block alert title. %s: The localized block name */
+		title: __( "'%s' needs an updated version of WordPress" ),
+		description: __( 'Please update to version 5.9 or above' ),
+	},
+};
```

Comment:
Probably an edge case but if for any other reason the Gallery block is missing, using this message could be misleading to the user.

In [this PR](https://github.com/WordPress/gutenberg/pull/46215/files#diff-f7e97c6adf52713aec986e0817c301fd3bb0d904074f105268f469f35483f5ecR154-R155) I explored passing the block name to the `native.missing_block_detail` filter hook, this way we could override the details string by block. We could use a similar approach here and, instead of setting the custom message here, use the filter hook in the [`galleryCheck` function](https://github.com/WordPress/gutenberg/pull/46157/files#diff-3286cc974baacc1ef30a6c7d6c18ae3ea400c86e1223c88029c1ea15051d65ecR201-R206) and based on the value of `galleryWithImageBlocks` override the string. For this approach, we'd also need to define a new filter hook to allow overriding the sheet title. WDYT?